### PR TITLE
feat: Timeline component

### PR DESCRIPTION
**Pull request by langz**
## Summary

Link to Figma: https://www.figma.com/file/cdtwQD8IJ7pTeE45U148r1/Eufemia---Web?node-id=19517%3A7350

### CODE REVIEW

**Comment by langz** [comment]
Will do some review after new years 🎆

File: packages/dnb-eufemia/src/components/timeline/TimelineItem.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  /**
+   * Text displaying the title of the item's corresponding page.
+   * Default: translations based on the icon.
+   */
+  icon_alt?: string
+
+  /**
+   * Text displaying the name of the timeline item.
+   */
+  name: string
```
Comment (thaytharma) [inline packages/dnb-eufemia/src/components/timeline/TimelineItem.tsx:35]: Should we consider using React.ReactNode instead here? 🤔

Comment (thaytharma) [reply packages/dnb-eufemia/src/components/timeline/TimelineItem.tsx:35]: Same for the other string props (date, info)

File: packages/dnb-design-system-portal/src/docs/uilib/components/timeline/demos.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+### Setting property 'state'
+
+Property `state` changes styling of icon, border, and font of the [timeline item](/uilib/components/timeline/properties#timelineitem-properties)
+
+<TimelineStates />
+
+### Setting property 'icon'
+
+Property `icon` is by default set based on the `state` property, but can be overridden by passing a `icon`.
```
Comment (thaytharma) [inline packages/dnb-design-system-portal/src/docs/uilib/components/timeline/demos.md:43]:
```suggestion
Property `icon` is by default set based on the `state` property, but can be overwritten by passing a `icon`.
```

File: packages/dnb-design-system-portal/src/docs/uilib/components/timeline/info.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+showTabs: true
+---
+
+## Description
+
+A timeline shows events in chronological order and gives a great overview of the overall process.
+The component itself has interchangable icons, additional messagebox when needed, and a step has three states(completed, current or upcoming). Date formatting can vary to be consistent with the rest of your application.
```
Comment (thaytharma) [inline packages/dnb-design-system-portal/src/docs/uilib/components/timeline/info.md:8]:
```suggestion
The component itself has interchangeable icons, additional messagebox when needed, and a step has three states(completed, current or upcoming). Date formatting can vary to be consistent with the rest of your application.
```

**Comment by thaytharma** [comment]
Awsome work, looks great! @langz 👏 ⭐

**Comment by langz** [comment]
> Will do some review and testing after new years 🎆
> 
> * Handle very long texts gracefully, especially the name prop:
>   image
> * Look at it/test it together with Casper.
> * Consider if the info property should rather be "children" and could be "anything".
>   But that sort of depends on how we see the use cases of the Timeline component.
>   Personally, I like too limit it as much as possible. But will check with Casper.
> * Refactor styling to use theme.

Done

**Comment by langz** [comment]
image

As of now the component TimelineItem has the following 3 properties:
- **name** (_Completed event_)
- **date** (_10. september 2021_)
- **info_message** (_Additional information about this step if needed._)

Maybe I should rename **name** to **title**, and **date** to **subtitle**? 🤔 
I've also considered changing **info_message** to **children** so that we could pass anything, but I've dropped that for now, as Casper and I think that's a bit too flexible as of now, and that info message is the only use case we know of for now.

File: packages/dnb-design-system-portal/src/docs/uilib/components/timeline.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+---
+title: 'Timeline'
+description: 'The Timeline component shows events in chronological order and gives a great overview of the overall process'
+status: 'new'
+order: 19
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/components/timeline.md:5]: I am not 100% sure about the value here.
I see a few components have the same value, others have a unique value.

**Comment by langz** [comment]
> Awesome work – again!
> 
> For new components, you can use safely use camelCase – if you wish? e.g. `info_message` => `infoMessage`

Aha, yeah I see that https://eufemia.dnb.no/uilib/components/breadcrumb/properties use camelCase as well.
I'll change that now.

**Pull request merged**